### PR TITLE
Update newlib and LLVM/Spike to RVV1.0

### PR DESCRIPTION
**Pull request by mp-17**
Update LLVM to version `15.0.0` and Spike to `1.1.1-dev`.
This updates the RVV support to 1.0 frozen.

### Fixed
 
- Fix wrong variable in `vmerge` and `vmv` `riscv-tests`

### Changed

- Update LLVM to version `15.0.0` (RVV 1.0)
- Update Spike to version `1.1.1-dev` (RVV 1.0)
- Update `newlib` from commit `84d068` to `5192d5`

## Checklist

- [x] Automated tests pass
- [x] Changelog updated
- [x] Code style guideline is observed

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -28,6 +28,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
  - Fixed AXI-inval-filter policy for D$ lines invalidation upon vector stores that are misaligned w.r.t. the D$ line width
  - Fix lane sequencer checks for floating-point comparisons
  - Fix synthesis error occuring due to the continuous assignmnet in the always block of mask unit
+ - Fix wrong variable in `vmerge` and `vmv` `riscv-tests`
```
Comment (suehtamacv) [inline CHANGELOG.md:31]: That is not part of this PR

**Review by suehtamacv — APPROVED** [review]
LGTM, can we merge this?

File: CHANGELOG.md (hunk 1 above)
Comment (mp-17) [reply CHANGELOG.md:31]: It can be considered part of the update since they fixed an LLVM bug and made `spike` crash without the fix. Without this, the CI won't pass ;-)

**Comment by mp-17** [comment]
I manage to make LLVM compile `newlib` and I should update the changelog accordingly.
Let the CI finish, then I can modify the changelog and merge immediately

**Comment by mp-17** [comment]
To me, the CI is clean. 
The (pull_request) flow is clean, and the (push) one failed because of a missing artifact to be downloaded (https://github.com/pulp-platform/ara/runs/7519958785?check_suite_focus=true). There is maybe a problem with the communication among the runners.
